common-app-lib/src/cmr/common_app/api/request_logger.clj

```diff
@@ -116,55 +136,60 @@
+   function multiple times in the routes list, then pass in an ID (the two arity call) and the
+   output will include the ID value in the log so that log entries can be told apart. One may want
+   to do this to see when a value logged becomes valid in the routes chain and thus availible
+   downstream.
+
+   NOTE: If you pass in an ID of :ignore then no ID will be written out, the same behavior if you
+   call the single arity call."
   ([handler]
    (log-ring-request handler :ignore-id))
   ([handler id]
    (fn [request]
      (if-not (config/enable-enhanced-http-logging)
        (handler request)
-       (let [start (tk/now-ms)
+       (let [response (handler request) ;; Do all the response handlers first
+             log-start (tk/now-ms) ;; After processing the other handlers, start tracking log time
+             _ (when-not (:ring-start-time request) ;; Did someone forget to setup a routes.clj
+                 (error "There is no ring start time for this service" (request->uri request)))
              now (dtp/clj-time->date-time-str (tk/now))
-             response (handler request)
+             ring-start (get request :ring-start-time (tk/now-ms))
              query-params (params/assoc-query-params request "UTF-8")
              form-params (params/assoc-form-params request "UTF-8")
-             ;; If adding or removing elements, change the log-version number so
-             ;; that reporting scripts can try to protect against change. Humans
-             ;; may be creating Splunk or ELK reports based on this content.
-             note (-> {"log-type" "action-log" "log-version" "1.0.0"}
+             note (-> {"message-type" "request-log"}
                       (add-hash-to-data response)
                       ;; As this handler can be called multiple times, if so,
                       ;; include an id showing which instance is reporting this
                       ;; information.
-                      (as-> data (if (= id :ignore-id) (assoc data "log-id" id) data))
+                      (as-> data (if (= id :ignore-id) data (assoc data "log-id" id)))
                       ;; assume that (add-body-hashes) has been run and reuse data
-                      (assoc "client-id" (get-in request [:headers "client-id"] "unknown")
+                      (into (:headers request)) ;; Bring in all the existing headers
+                      (assoc "cmr-hit" (get-in response [:headers "CMR-Hit"] "n/a")
+                             "cmr-took" (get-in response [:headers "CMR-Took"] "n/a")
                              "form-params" (dump-param form-params :form-params)
                              "method" (string/upper-case
                                        (name (get-in request [:request-method] "unknown")))
                              "now" now
-                             "protocol" (:protocol request)
+                             "now-ms" log-start
```

Comment:
now is a string, human readable, now-ms is a number you can do math with in splunk